File: app/src/main/java/com/example/comiaseokt/api/CategoriaApi.java
package com.example.comiaseokt.api;

import com.example.comiaseokt.entity.GenericResponse;
import com.example.comiaseokt.entity.service.Categoria;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;

public interface CategoriaApi {
    String base = "api/categoria";

    @GET(base)
    Call<GenericResponse<List<Categoria>>> listarCategoriasActivas();
}

File: app/src/main/java/com/example/comiaseokt/api/CategoriaApi.kt
package com.example.comiaseokt.api

import com.example.comiaseokt.activity.GenericResponse
import com.example.comiaseokt.entity.service.Categoria
import retrofit2.Call
import retrofit2.http.GET

interface CategoriaApi {
    @GET(base)
    fun listarCategoriasActivas(): Call<GenericResponse<List<Categoria?>?>?>?

    companion object {
        const val base = "api/categoria"
    }
}
